openpower/package/pkg-versions.mk

```diff
@@ -0,0 +1,137 @@
+################################################################################
+#
+# pkg-versions
+#
+# Read VERSION.readme in the current directory to learn about the version
+# string structure
+#
+################################################################################
+
+define OPENPOWER_SUBPACKAGE_VERSION
+
+$(2)_VERSION_FILE = $$(OPENPOWER_VERSION_DIR)/$(1).version.txt
+ALL_SUBPACKAGE_VERSIONS += $$($(2)_VERSION_FILE)
+
+### Create subpackage patch file
+define $(2)_OPENPOWER_PATCH_FILE
+
+mkdir -p "$$(OPENPOWER_VERSION_DIR)"
+
+cd "$$(PKGDIR)"; if ls *.patch; then sha512sum *.patch | sha512sum | cut -c 1-7 | \
```

Comment:
1) BR2_GLOBAL_PATCH_DIR is a list.   So you're going to need to have a
foreach.  Do you really need to 'cd' into the BR2_GLOBAL_PATCH_DIR though?
Why can't you just 'ls *.patch $$(foreach
path,$$(BR2_GLOBAL_PATCH_DIR),$path/$(1))'

2) There might be an issue with how I defined BR2_GLOBAL_PATCH_DIR.  Check
out pkg-generic.mk in buildroot and see how they use it.  I recently added
a BR2_GLOBAL_PATCH_DIR+=... in openpower/external.mk and maybe I did it
wrong.

On Sun, Mar 8, 2015 at 2:28 PM, Stephen Cprek <notifications@github.com>
wrote:

> In openpower/package/pkg-versions.mk
> <https://github.com/open-power/op-build/pull/58#discussion_r26008054>:
>
> > +# Read VERSION.readme in the current directory to learn about the version
> > +# string structure
> > +#
> > +################################################################################
> > +
> > +define OPENPOWER_SUBPACKAGE_VERSION
> > +
> > +$(2)_VERSION_FILE = $$(OPENPOWER_VERSION_DIR)/$(1).version.txt
> > +ALL_SUBPACKAGE_VERSIONS += $$($(2)_VERSION_FILE)
> > +
> > +### Create subpackage patch file
> > +define $(2)_OPENPOWER_PATCH_FILE
> > +
> > +mkdir -p "$$(OPENPOWER_VERSION_DIR)"
> > +
> > +cd "$$(PKGDIR)"; if ls *.patch; then sha512sum *.patch | sha512sum | cut -c 1-7 | \
>
> I cannot figure out why there are extra double quotes in
> BR2_GLOBAL_PATCH_DIR
> cd $$(BR2_GLOBAL_PATCH_DIR)/$(1); ls;
> and cd "$$(BR2_GLOBAL_PATCH_DIR)/$(1)"; ls;
> give the below result
> cd "" "
> /esw/san2/smcprek/open-power/forks/op-build/openpower/custom/patches
> "/hostboot; ls;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/open-power/op-build/pull/58/files#r26008054>.
>